testscript/testscript.go

```diff
@@ -478,6 +485,33 @@ func (ts *TestScript) execBackground(command string, args ...string) (*exec.Cmd,
 	return cmd, cmd.Start()
 }
 
+func (ts *TestScript) buildExecCmd(command string, args ...string) (*exec.Cmd, error) {
+	if filepath.Base(command) == command {
+		if lp, err := ts.lookPath(command); err != nil {
+			return nil, err
```

Comment:
Why not just let `PATH=` give the same error that `PATH=emptydir` would? You'd also make the code simpler, and not break with the `exec.Command` signature.

Reply:
Per our Slack chat; we need to eagerly use `TestScript`'s `lookPath` here, else `os/exec`'s `LookPath` will be used as part of `os/exec.Command` and then set a non-exported `error` field which is then tripped during `Start/Run` (https://github.com/golang/go/blob/006a5e7d00992cfae6ac406959512d680025f75c/src/os/exec/exec.go#L164-L177).